source/common/dynamo/dynamo_request_parser.cc

```diff
@@ -60,60 +59,47 @@ std::string RequestParser::parseOperation(const Http::HeaderMap& headerMap) {
+                                                         const Json::Object& json_data) {
   TableDescriptor table{"", true};
 
   // Simple operations on a single table, have "TableName" explicitly specified.
   if (find(SINGLE_TABLE_OPERATIONS.begin(), SINGLE_TABLE_OPERATIONS.end(), operation) !=
       SINGLE_TABLE_OPERATIONS.end()) {
-    Json::StringLoader json(data);
-    if (json.hasObject("TableName")) {
-      table.table_name = json.getString("TableName");
-    }
+    table.table_name = json_data.getString("TableName", "");
   } else if (find(BATCH_OPERATIONS.begin(), BATCH_OPERATIONS.end(), operation) !=
              BATCH_OPERATIONS.end()) {
-    Json::StringLoader json(data);
-    if (json.hasObject("RequestItems")) {
-      Json::Object tables = json.getObject("RequestItems");
-      tables.iterate([&table](const std::string& key, const Json::Object&) {
-        if (table.table_name.empty()) {
-          table.table_name = key;
-        } else {
-          if (table.table_name != key) {
-            table.table_name = "";
-            table.is_single_table = false;
-            return false;
-          }
+    Json::Object tables = json_data.getObject("RequestItems", true);
+    tables.iterate([&table](const std::string& key, const Json::Object&) {
+      if (table.table_name.empty()) {
+        table.table_name = key;
+      } else {
+        if (table.table_name != key) {
+          table.table_name = "";
+          table.is_single_table = false;
+          return false;
         }
-
-        return true;
-      });
-    }
+      }
+      return true;
+    });
   }
 
   return table;
 }
-std::vector<std::string> RequestParser::parseBatchUnProcessedKeys(const std::string& data) {
+std::vector<std::string> RequestParser::parseBatchUnProcessedKeys(const Json::Object& json_data) {
   std::vector<std::string> unprocessed_tables;
-  Json::StringLoader json(data);
-  if (json.hasObject("UnprocessedKeys")) {
-    Json::Object tables = json.getObject("UnprocessedKeys");
-    tables.iterate([&unprocessed_tables](const std::string& key, const Json::Object&) {
-      unprocessed_tables.emplace_back(key);
-      return true;
-    });
-  }
+  Json::Object tables = json_data.getObject("UnprocessedKeys", true);
+  tables.iterate([&unprocessed_tables](const std::string& key, const Json::Object&) {
+    unprocessed_tables.emplace_back(key);
+    return true;
+  });
+
   return unprocessed_tables;
 }
-std::string RequestParser::parseErrorType(const std::string& data) {
-  Json::StringLoader json(data);
-
-  if (json.hasObject("__type")) {
-    std::string error_type = json.getString("__type");
-    for (const std::string& supported_error_type : SUPPORTED_ERROR_TYPES) {
-      if (StringUtil::endsWith(error_type, supported_error_type)) {
-        return supported_error_type;
-      }
+std::string RequestParser::parseErrorType(const Json::Object& json_data) {
+  std::string error_type = json_data.getString("__type", "");
```

Comment:
if __type is not set on the json, we'll iterate though all supported_error_types.

we could early exit if __type is missing or if error_type after json_data.getString("__type", ""); is empty.